src/pages/SingleVideo/SingleVideo.js

```diff
@@ -1,13 +1,108 @@
 import "./single-video.css";
 import { Sidebar } from "../../components/Sidebar/Sidebar";
 import ReactPlayer from "react-player";
-import { useParams } from "react-router-dom";
+import {useNavigate, useParams } from "react-router-dom";
 import { useVideos } from "../../contexts/VideosContext";
 import { getVideoLink } from "../../utils/getVideoLink";
+import { useWatchLater } from "../../contexts/WatchLaterContext";
+import {
+  removeFromWatchLaterService,
+  addToWatchLaterService,
+} from "../../services/watchLaterService";
+import { useAuth } from "../../contexts/AuthContext";
+import { useLike } from "../../contexts/LikeContext";
+import { useState } from "react";
+import { PlaylistModal } from "../../components/PlaylistModal/PlaylistModal";
+import {
+  addLikedVideoService,
+  removeLikedVideoService,
+} from "../../services/likeServices";
+
 const SingleVideo = () => {
   const { videoId } = useParams();
   const { videos } = useVideos();
+  const navigate = useNavigate();
+  const [playlistModal, setPlaylistModal] = useState(false);
+
   const video = videos.find((v) => v._id === videoId);
+  const {
+    watchLaterState: { watchLaterVideos },
+    dispatchWatchLater,
+  } = useWatchLater();
+
+  const {
+    likeState: { likedVideos },
+    dispatchLike,
+  } = useLike();
+
+  const videoInWatchLater = watchLaterVideos.find(
```

Comment:
This is a good utility. Useful in many places. Good that you created a function.